packages/ui-core/src/components/Search/Search.tsx

```diff
@@ -0,0 +1,172 @@
+    changeDelay?: number;
+    /** Change handler */
+    onChange?: (value: string) => void;
+    /** Submit handler */
+    onSubmit?: (value: string) => void;
+}
+
+const cn = cnCreate('mfui-search');
+const Search: React.FC<ISearchProps> = ({
+        value = '',
+        placeholder,
+        hideIcon,
+        items = [],
+        changeDelay = 250,
+        onChange,
+        onSubmit,
+}) => {
+    const [searchQuery, setSearchQuery] = useState(value);
+    const [activeIndex, setActiveIndex] = useState(-1);
+    const [isFocused, setFocus] = useState(false);
+    const debouncedOnChange = useRef(debounce((inputValue) => onChange && onChange(inputValue), changeDelay));
+
+    useEffect(() => setSearchQuery(value), [value, setSearchQuery]);
+
+    const handleChange: React.EventHandler<React.ChangeEvent<HTMLInputElement>> = (e) => {
+        const { target: { value: inputValue = '' } } = e;
+
+        setSearchQuery(inputValue);
+        setActiveIndex(-1);
+
+        changeDelay === 0 ? onChange && onChange(inputValue) : debouncedOnChange.current(inputValue);
+    };
+
+    const handleHoverItem = useCallback((index: number) => (_e: React.SyntheticEvent<EventTarget>): void => {
+        setActiveIndex(index);
+    }, [setActiveIndex]);
+
+    const handleSearchSubmit: HandleSearchSubmit = useCallback((): void => {
+        onSubmit && searchQuery && onSubmit(searchQuery);
+    }, [searchQuery, onSubmit]);
+
+    const handleItemSubmit: HandleItemSubmit = useCallback((index: number): void => {
+        const chosenValue = items[index];
+
+        onSubmit && onSubmit(chosenValue);
+    }, [onSubmit, items]);
+
+    const handleSelectSubmit: HandleSelectSubmit =
+        useCallback(() => (_e): void => {
+            handleItemSubmit(activeIndex);
+        }, [handleItemSubmit, activeIndex]);
+
+    const handleFieldFocusToggle = useCallback((): void => {
+        setFocus(focus => !focus);
+    }, [setFocus]);
+
+    const handleClick = useCallback((): void => {
+        if (activeIndex >= 0) {
+            setActiveIndex(-1);
+        }
+    }, [activeIndex, setActiveIndex]);
+
+    const handleKeyDown = useCallback((e: React.KeyboardEvent<HTMLDivElement>): boolean => {
+        if (e.key === 'ArrowDown' && activeIndex < items.length - 1) {
+            setActiveIndex(index => index + 1);
+            e.preventDefault();
+        } else if (e.key === 'ArrowUp' && activeIndex > -1) {
+            setActiveIndex(index => index - 1);
+            e.preventDefault();
+        } else if (e.key === 'Enter' && activeIndex > -1) {
+            handleItemSubmit(activeIndex);
+        } else if (e.key === 'Enter' && activeIndex === -1) {
+            handleSearchSubmit();
+        }
+
+        return false;
+    }, [activeIndex, setActiveIndex, handleSearchSubmit, handleItemSubmit]);
+
+    const highlightString = (title) => {
+        const stringFragments = title.split(RegExp(`(${searchQuery})`, 'ig'));
+
+        return (
+            <div>
```

Comment:
Тут даже div не нужен вроде как
```suggestion
            <>
```